### PR TITLE
refactor: Make the end-to-end test a bit more manageable

### PR DESCRIPTION
**Pull request by carols10cents**
When going to add tests for the Flight API to the end-to-end tests, I found it really hard to navigate the end-to-end test that's getting quite long. I've extracted code into smaller pieces here to make it so that you have to look through less code to see what's going on, see what parts depend on other parts, and find where you need to change code.

I also split the code from the test file out into multiple files, and I had to use the `mod.rs` naming pattern I dislike because otherwise the module file in `tests/` would get run as if it were an integration test crate.

This still isn't the greatest ever, but I think it's better. I'll be building on this with the Flight tests.

### CODE REVIEW

File: tests/end-to-end.rs [hunk 1]
```diff
@@ -48,153 +37,212 @@ const TOKEN: &str = "InfluxDB IOx doesn't have authentication yet";
...
+    grpc_api::read_window_aggregate_test(&http_client, &influxdb2, &mut storage_client).await;
+    test_http_error_messages(&influxdb2).await.unwrap();
+}
+
+// TODO: Randomly generate org and bucket ids to ensure test data independence
+// where desired
+
+#[derive(Debug)]
+pub struct Scenario {
+    org_id_str: String,
```
Comment (alamb) [inline tests/end-to-end.rs:78]: 💯 🏅

**Review by alamb — APPROVED** [review]
I really like these changes @carols10cents  -- thank you for taking the time to clean it up!

It would be nice, though not required, to pull the cargo dep cleanup into its own PR, to try and help me with the conflict this will cause with #707  but overall 👍 really nice work

File: Cargo.toml [hunk 2]
```diff
@@ -34,6 +34,8 @@ debug = true
 debug = true
 
 [dependencies]
+
```
[the remark this answers is not in the document]
Comment (carols10cents) [reply Cargo.toml]: No problem, done!

File: tests/end-to-end.rs [hunk 3]
```diff
@@ -48,153 +37,212 @@ const TOKEN: &str = "InfluxDB IOx doesn't have authentication yet";
...
 
-    let org_id_str = "0000111100001111";
-    let org_id = u64::from_str_radix(org_id_str, 16).unwrap();
-    let bucket_id_str = "1111000011110000";
-    let bucket_id = u64::from_str_radix(bucket_id_str, 16).unwrap();
+    let http_client = reqwest::Client::new();
+    let influxdb2 = influxdb2_client::Client::new(HTTP_BASE, TOKEN);
+    let mut storage_client = StorageClient::connect(GRPC_URL_BASE).await.unwrap();
+
+    // These tests share data; TODO: a better way to indicate this
```
[the remark this answers is not in the document]
Comment (carols10cents) [reply tests/end-to-end.rs:52]: Yeah, I wasn't sure how much we wanted to optimize. And I couldn't remember if we wanted to test the `influxdb2_client` lib as well, as we are indirectly by using it to build points in the current `load_data` function... I think having a `Scenario::load_data` fn that takes line protocol would be useful. Future tho.

**Pull request merged**